### PR TITLE
Update location-retrieval for lastLocationTime documentation cardinality

### LINKED ISSUE #197: Consistency problem in Location retrieval regarding lastLocationTime cardinality
**Problem description**
Seem that we have still a consistency problem in Location retrieval.

API documentation indicates: `In this case the system may still return lastLocationTime, if available.` see line 40

While the attribute `lastLocationTime`  is mandatory in location retrieval response

**Expected behavior**

Shift  `lastLocationTime`  to optional

**Alternative solution**
Change documentation to _may_ to _must_

**Additional context**

### PR DESCRIPTION
**Pull request by bigludo7**
#### What type of PR is this?

Add one of the following kinds:

* correction

#### What this PR does / why we need it:

Align the documentation part of the API with the fact that `lastLocationTime` is mandatory in all response.
See line 40

#### Which issue(s) this PR fixes:

Fixes #197 

#### Changelog input

```
 release-note

```

#### Additional documentation 

This section can be blank.

```
docs

```

### CODE REVIEW

File: code/API_definitions/location-retrieval.yaml [hunk 1]
```diff
@@ -37,7 +37,7 @@ info:
 
     * **Max Age**: Maximum age of the location information which is accepted for the location retrieval (in seconds).
 
-      * Absence of maxAge means "any age" is acceptable for the client. In other words, this is like maxAge=infinite. In this case the system may still return lastLocationTime, if available. If the system is not able to provide location a error 404 with code LOCATION_RETRIEVAL.DEVICE_NOT_FOUND is send back.
+      * Absence of maxAge means "any age" is acceptable for the client. In other words, this is like maxAge=infinite. The system still return lastLocationTime in the response. If the system is not able to provide location a error 404 with code LOCATION_RETRIEVAL.DEVICE_NOT_FOUND is send back.
```
Comment (alpaycetin74) [inline code/API_definitions/location-retrieval.yaml:40]: Could we change the wording as either "the system will return" or "the system returns" ? Thank you.

Comment (bigludo7) [reply code/API_definitions/location-retrieval.yaml]: @alpaycetin74 Sure !
done

File: code/API_definitions/location-retrieval.yaml [hunk 2]
```diff
@@ -37,7 +37,7 @@ info:
 
     * **Max Age**: Maximum age of the location information which is accepted for the location retrieval (in seconds).
 
-      * Absence of maxAge means "any age" is acceptable for the client. In other words, this is like maxAge=infinite. In this case the system may still return lastLocationTime, if available. If the system is not able to provide location a error 404 with code LOCATION_RETRIEVAL.DEVICE_NOT_FOUND is send back.
+      * Absence of maxAge means "any age" is acceptable for the client. In other words, this is like maxAge=infinite. The system returns lastLocationTime in the response. If the system is not able to provide location a error 404 with code LOCATION_RETRIEVAL.DEVICE_NOT_FOUND is send back.
```
Comment (jlurien) [inline code/API_definitions/location-retrieval.yaml:40]:
```suggestion
      * Absence of `maxAge` means that "any age" is acceptable for the client. In other words, this is like `maxAge`=infinite. The system will return `lastLocationTime` in the response. If the system is not able to provide location, an error 404 with code LOCATION_RETRIEVAL.DEVICE_NOT_FOUND is sent back.
```
Also, we should fix the line below to:

    * `maxAge`=0 means that a fresh calculation is requested by the client. If the system is not able to provide the fresh location, an error 422 with code LOCATION_RETRIEVAL.UNABLE_TO_FULFILL_MAX_AGE is sent back.

**Review by alpaycetin74 — APPROVED** [review]
I approve, and I also agree with @jlurien 's comment. Thank you.

**Review by jlurien — APPROVED** [review]
LGTM, thanks

**Pull request merged**